### PR TITLE
[Robot] RD2 testcases

### PR DESCRIPTION
**Pull request by rjanjanam**
Enhanced recurring donations(RD2) has to be enabled first if it is not enabled which will deploy the new Ui fields relevant. Try to create an RD of type Fixed and open using the new modal. Validate the respective sections
# Critical Changes

# Changes

# Issues Closed

# Community Ideas Delivered

# New Metadata

# Deleted Metadata

### CODE REVIEW

File: robot/Cumulus/resources/NPSP.py [hunk 1]
```diff
@@ -997,7 +1001,8 @@ def click_link_with_text(self, text):
         locator = npsp_lex_locators['link-text'].format(text)
         self.selenium.wait_until_page_contains_element(locator)
         element = self.selenium.driver.find_element_by_xpath(locator)
-        self.selenium.driver.execute_script('arguments[0].click()', element)  
+        self.selenium.driver.execute_script('arguments[0].click()', element)
+        time.sleep(1)
```
Comment (sgovindankutty) [inline robot/Cumulus/resources/NPSP.py:1005]: Do you want to wait on some elements instead of sleep?

File: robot/Cumulus/resources/NPSP.py [hunk 2]
```diff
@@ -1548,4 +1553,30 @@ def get_url_formatted_object_name(self,name):
...
+        if self.npsp.check_if_element_displayed(locator):
+            isPresent = True
+        return isPresent
+
+    def check_rd2_is_enabled(self):
+        """Verifies that Enhanced Recurring Donations is enabled on the org"""
+        enabled = False
+        if self.check_submenu_link_exists("Upgrade to Enhanced Recurring Donations"):
+            self.click_link_with_text("Upgrade to Enhanced Recurring Donations")
+            time.sleep(2)
```
Comment (sgovindankutty) [inline robot/Cumulus/resources/NPSP.py:1578]: Can this sleep be avoided?

File: robot/Cumulus/resources/NPSP.py [hunk 3]
```diff
@@ -1548,4 +1553,30 @@ def get_url_formatted_object_name(self,name):
         object_name = "{}{}".format(self.cumulusci.get_namespace_prefix(), name)
         out['baseurl'] = base_url
         out['objectname'] = object_name
-        return out
\ No newline at end of file
+        return out
+    
+    def check_submenu_link_exists(self,title):
```
[the remark this answers is not in the document]
Comment (rjanjanam) [reply robot/Cumulus/resources/NPSP.py:1558]: adding

File: robot/Cumulus/resources/locators_49.py [hunk 4]
```diff
@@ -258,6 +258,16 @@
         'form_object_dropdown':'//*[@data-qa-locator="{}"]//button',
         'object_field_checkbox':'//*[@data-qa-locator="{}"]//span[@class="slds-checkbox_faux"]',
         },
+    'erd':{
```
[the remark this answers is not in the document]
Comment (rjanjanam) [reply robot/Cumulus/resources/locators_49.py:277]: added

**Pull request merged**